src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -2616,7 +2616,7 @@ internal class MethodCacheEntry : CacheEntry
             /// </summary>
             internal Func<string, DotNetAdapter, object, DotNetAdapter.MethodCacheEntry, bool, bool, PSMethod> PSMethodCtor;
 
-            internal MethodCacheEntry(MethodBase[] methods)
+            internal MethodCacheEntry(IList<MethodBase> methods)
```

Comment:
Use `IList<T>` will avoid creating an array with `.ToArray()` when calling this constructor.

The cache table won't benefit much from pre-building in a source generator because it's lazy populated, only when a user attempts to access any member on any types. It doesn't have a fixed set of types that always need to be populated.

Reply:
> Use IList<T> will avoid creating an array with .ToArray() when calling this constructor.

My question is about performance. It seems all callsites use List<MethodBase> and we could use the type here too 

>  It doesn't have a fixed set of types that always need to be populated.

My thoughts is about .Net types. We pre-build type cache for them and could do the same for its members. I guess it is most common scenario to access .Net type's members.

Reply:
I don't think there is a noticeable difference between using `IList<MethodBase>` and `List<MethodBase>` as the parameter type.

> My thoughts is about .Net types.

I was talking about .NET types too. There is no fixed set of types that will always be needed. If you mean populate all .NET types, that would be a burden to the memory usage.